### PR TITLE
Fixed e2e test reliability, added Debian9 target

### PR DESCRIPTION
**Pull request by AhmedBM**
## Description

* Removed static `aziot-identity-service_1.2.3` install on Ubuntu 20.04, which would randomly fail on dpkg issues during install. Replaced with installing the latest stable version on packages.microsoft.com
* Onboarded Debian 9 as additional test target
* Removed Azure policy to auto-shutdown VM in favor of pipelines deallocating VMs after tests

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] I added unit-tests to validate my changes. All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: devops/terraform/host/OSConfigHost.tf [hunk 1]
```diff
@@ -182,17 +182,3 @@ resource "azurerm_linux_virtual_machine" "osconfigvm" {
         environment = var.environment_tag
     }
 }
-
-resource "azurerm_dev_test_global_vm_shutdown_schedule" "vm_shutdown_schedule" {
```
Comment (robertschaedler3) [inline devops/terraform/host/OSConfigHost.tf:186]: Does this change anything related to the cleanup of E2E resources?

File: devops/pipeline/e2etest.yaml [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # E2E test pipeline - performs end-to-end tests on the following platforms
+#  - Debian 9
 #  - Ubuntu 18.04
 #  - Ubuntu 20.04
+# 
+# See "!!! ADD NEW DISTROS HERE !!!" for new distro placeholder
```
Comment (MariusNi) [inline devops/pipeline/e2etest.yaml:7]: Is this comment really necessary?

If it is, please remove the many ! and change to lowerase

File: devops/pipeline/e2etest.yaml [hunk 3]
```diff
@@ -115,14 +118,20 @@ stages:
...
           # See https://docs.microsoft.com/en-us/azure/virtual-machines/linux/cli-ps-findimage
-          # debian-9:
-          #   distroName: debian-9
-          #   image_publisher: credativ
-          #   image_offer: Debian
-          #   image_sku: 9-backports
-          #   image_version: 9.20210721.0
-          #   packagePattern: '**/*stretch_x86_64.deb'
-          #   device_id: debian9
+          # !!! ADD NEW DISTROS HERE !!!
```
Comment (MariusNi) [inline devops/pipeline/e2etest.yaml:121]: > # !!! ADD NEW DISTROS HERE !!! [](http://example.com/codeflow?start=9&length=31)

Same as earlier. Remove the many ! and change to lowercase

File: devops/pipeline/e2etest.yaml [hunk 4]
```diff
@@ -213,9 +228,10 @@ stages:
   - job: E2ETestJob
     strategy:
       matrix:
-        # debian-9:
-        #     distroName: debian-9
-        #     device_id: debian9
+        # !!! ADD NEW DISTROS HERE !!!
```
Comment (MariusNi) [inline devops/pipeline/e2etest.yaml:231]: > # !!! ADD NEW DISTROS HERE !!! [](http://example.com/codeflow?start=7&length=31)

Same as for the other instances. Will stop commenting for this issue here.

File: devops/terraform/host/OSConfigHost.tf (hunk 1 above)
Comment (AhmedBM) [reply devops/terraform/host/OSConfigHost.tf:186]: no, same azure resources, this is just an Azure Policy on the hosts we create

File: devops/pipeline/e2etest.yaml (hunk 2 above)
Comment (AhmedBM) [reply devops/pipeline/e2etest.yaml:7]: Its meant to be a placeholder to we can just find the different placeholder locations and add the new distro into those placeholders

File: devops/pipeline/e2etest.yaml (hunk 3 above)
Comment (AhmedBM) [reply devops/pipeline/e2etest.yaml:121]: im trying to make it obvious that its a placeholder, open to suggestions

File: devops/pipeline/e2etest.yaml [hunk 5]
```diff
@@ -1,6 +1,9 @@
 # E2E test pipeline - performs end-to-end tests on the following platforms
+#  - Debian 9
 #  - Ubuntu 18.04
 #  - Ubuntu 20.04
+# 
+# See "!!! ADD NEW DISTROS HERE !!!" for new distro placeholder
```
Comment (MariusNi) [inline devops/pipeline/e2etest.yaml:6]: >See "!!! ADD NEW DISTROS HERE !!!" for new distro placeholder [](http://example.com/codeflow?start=2&length=61)

How about: "Add new distros here as needed:"

Comment (MariusNi) [reply devops/pipeline/e2etest.yaml:6]: No lots of !s, no all uppercase...

Comment (MariusNi) [reply devops/pipeline/e2etest.yaml:6]: This comment here does not appear necessary. The others in the body can switch to be normal, as suggested above.

**Review by MariusNi — APPROVED** [review]
:shipit:

**Review by robertschaedler3 — APPROVED** [review]
👍🏻

**Pull request merged**